ngraph/core/src/op/transpose.cpp

```diff
@@ -78,56 +79,42 @@ shared_ptr<Node> op::v1::Transpose::clone_with_new_inputs(const OutputVector& ne
 
 namespace transpose
 {
-    template <element::Type_t ET>
-    std::vector<int64_t> get_vector(const HostTensorPtr& arg)
-    {
-        std::vector<int64_t> rc;
-        auto p = arg->get_data_ptr<ET>();
-        for (size_t i = 0; i < shape_size(arg->get_shape()); i++)
-        {
-            rc.push_back(p[i]);
-        }
-        return rc;
-    }
-
     bool evaluate_transpose(const HostTensorPtr& arg1,
                             const HostTensorPtr& arg2,
                             const HostTensorPtr& out)
     {
         NGRAPH_CHECK(arg2->get_element_type().is_integral_number(),
-                     "axis element type is not integral data type");
-
-        std::vector<int64_t> axis_order = host_tensor_2_vector<int64_t>(arg2);
+                     "Transpose axis element type is not integral data type");
 
+        std::vector<int64_t> axes_order = host_tensor_2_vector<int64_t>(arg2);
         Shape in_shape = arg1->get_shape();
-        AxisVector in_axis_order(shape_size(arg2->get_shape()));
-        if (in_axis_order.empty())
-        {
-            size_t rank = in_shape.size();
-            for (size_t i = 1; i <= rank; ++i)
-                in_axis_order.emplace_back(rank - i);
-        }
-        else
+        if (shape_size(arg2->get_shape()) == 0)
         {
-            std::transform(axis_order.begin(),
-                           axis_order.end(),
-                           in_axis_order.begin(),
-                           [&](const int64_t& v) { return (v > 0) ? v : 0; });
+            axes_order.resize(in_shape.size());
+            std::iota(axes_order.begin(), axes_order.end(), 0);
+            std::reverse(axes_order.begin(), axes_order.end());
         }
+        bool is_unique_order =
+            (std::unique(axes_order.begin(), axes_order.end()) == axes_order.end());
```

Comment:
Ensure `axes_order` sorted or use `std::set` to check if indices are unique.

Reply:
Changed to `std::set` size check. Thanks!